### PR TITLE
feat(textfield): add end-alignment

### PR DESCRIPTION
**Pull request by asyncLiz**

### CODE REVIEW

File: packages/mdc-textfield/README.md [hunk 1]
```diff
@@ -355,6 +355,7 @@ Property | Value Type | Description
 `trailingIconAriaLabel` | `string` (write-only) | Proxies to the foundation's `setTrailingIconAriaLabel` method.
 `leadingIconContent` | `string` (write-only) | Proxies to the foundation's `setLeadingIconContent` method.
 `trailingIconContent` | `string` (write-only) | Proxies to the foundation's `setTrailingIconContent` method.
+`endAligned` | `boolean` | Aligns the input to the end if true, or start if false.
```
Comment (patrickrodee) [inline packages/mdc-textfield/README.md]: Let's add the class name to the "Classes" section (line 260)

File: packages/mdc-textfield/component.ts [hunk 2]
```diff
@@ -326,6 +326,25 @@ export class MDCTextField extends MDCComponent<MDCTextFieldFoundation> implement
     this.foundation_.setUseNativeValidation(useNativeValidation);
   }
 
+  /**
+   * Indicates if a text field is end-aligned if true, or start-aligned if false.
+   */
+  get endAligned() {
```
[the remark this answers is not in the document]
Comment (asyncLiz) [reply packages/mdc-textfield/component.ts]: Removed the prop

File: packages/mdc-textfield/README.md (hunk 1 above)
Comment (asyncLiz) [reply packages/mdc-textfield/README.md]: Added!

File: test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    <link rel="stylesheet" href="../../../out/spec/fixture.css">
+    <link rel="stylesheet" href="../../../out/spec/mdc-textfield/fixture.css">
+  </head>
+
+  <body class="test-container">
+  <main class="test-viewport test-viewport--mobile">
+    <div class="test-layout">
+
+      <div class="test-cell test-cell--textfield">
+        <label class="mdc-text-field mdc-text-field--end-aligned mdc-text-field--no-label">
```
Comment (patrickrodee) [inline test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html]: Should this be a `<div>`? Same for the rest of the examples.

File: test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+-->
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title>Baseline Text Field Element end-aligned - MDC Web Screenshot Test</title>
```
Comment (patrickrodee) [inline test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html]: Let's add an example for textarea.

File: packages/mdc-textfield/_mixins.scss [hunk 5]
```diff
@@ -808,6 +808,18 @@
   }
 }
 
+// End aligned
+@mixin mdc-text-field-end-aligned_ {
+  .mdc-text-field__input {
+    // IE11 does not support text-align: end
```
Comment (patrickrodee) [inline packages/mdc-textfield/_mixins.scss]: Of course not... oh well.

File: test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html (hunk 3 above)
Comment (asyncLiz) [reply test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html]: Woops, must've gotten copied over from the other work. I'll fix that

Comment (patrickrodee) [reply test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html]: YOU'RE TOO FAST 🏎

Comment (asyncLiz) [reply test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html]: #gottagofast

File: test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html (hunk 4 above)
Comment (asyncLiz) [reply test/screenshot/spec/mdc-textfield/classes/baseline-end-aligned.html]: Added textarea test!

**Comment by patrickrodee** [comment]
Looks like some flaky menu screenshots already got fixed. Might be able to revert the diffing.json updates by resolving merge conflicts.

**Pull request merged**